src/tir/transforms/thread_storage_sync.cc

```diff
@@ -113,7 +113,6 @@ class ThreadSyncPlanner : public StorageAccessVisitor {
         }
       }
       if (sync_before_stmt) {
-        ICHECK_EQ(condition_counter(), 0) << "Cannot insert syncs inside condition";
```

Comment:
we need to be careful this may result in a condition that threads can never reach the sync point if they diverge